src/util/RequestsUtil.ts

```diff
@@ -93,4 +95,32 @@ export class RequestsUtil {
 		}
 		return this.hashCode( resolver.tag ) & 0x00FFFFFF;
 	}
+
+	public static async checkTicketValidity( ticketKeyString: string ): Promise<boolean> {
+		try {
+			const searchResults = await MojiraBot.jira.issueSearch.searchForIssuesUsingJqlGet( {
+				jql: `(${ BotConfig.request.invalidRequestJql }) AND key in (${ ticketKeyString })`,
+				fields: ['key'],
+			} );
+			const invalidTickets = searchResults.issues.map( ( { key } ) => key );
+			if ( invalidTickets.length > 0 ) {
+				return false;
+			} else {
+				return true;
+			}
+		} catch ( err ) {
+			this.logger.error( err );
+			return false;
+		}
+	}
+
+	public static checkTicketLinks( content: string ): number {
+		const regex = new RegExp( `(?:${ MentionCommand.getTicketLinkRegex().source }|(${ MentionCommand.ticketPattern }))(\\?\\S+)?`, 'g' );
```

Comment:
The `MentionCommand` regexes should probably be extracted into their own util class. But out of scope for this PR.